app/components/autocomplete/emoji_suggestion/emoji_suggestion.tsx

```diff
@@ -0,0 +1,235 @@
+        }
+
+        return [];
+    }, [fuse, searchTerm]);
+
+    const showingElements = Boolean(data.length);
+
+    const completeSuggestion = useCallback((emoji: string) => {
+        const emojiPart = value.substring(0, cursorPosition);
+
+        if (emojiPart.startsWith('+:')) {
+            handleReactionToLatestPost(serverUrl, emoji, true, rootId);
+            updateValue('');
+            return;
+        }
+        if (emojiPart.startsWith('-:')) {
+            handleReactionToLatestPost(serverUrl, emoji, false, rootId);
+            updateValue('');
+            return;
+        }
+
+        // We are going to set a double : on iOS to prevent the auto correct from taking over and replacing it
+        // with the wrong value, this is a hack but I could not found another way to solve it
+        let completedDraft: string;
+        let prefix = ':';
+        if (Platform.OS === 'ios') {
+            prefix = '::';
+        }
+
+        const emojiData = getEmojiByName(emoji);
+        if (emojiData?.image && emojiData.category !== 'custom') {
+            const codeArray: string[] = emojiData.image.split('-');
+            const code = codeArray.reduce((acc, c) => {
+                return acc + String.fromCodePoint(parseInt(c, 16));
+            }, '');
+            completedDraft = emojiPart.replace(EMOJI_REGEX_WITHOUT_PREFIX, `${code} `);
+        } else {
+            completedDraft = emojiPart.replace(EMOJI_REGEX_WITHOUT_PREFIX, `${prefix}${emoji}: `);
+        }
+
+        if (value.length > cursorPosition) {
+            completedDraft += value.substring(cursorPosition);
+        }
+
+        updateValue(completedDraft);
+
+        if (Platform.OS === 'ios' && (!emojiData?.filename || emojiData.category !== 'custom')) {
+            // This is the second part of the hack were we replace the double : with just one
+            // after the auto correct vanished
+            setTimeout(() => {
+                updateValue(completedDraft.replace(`::${emoji}: `, `:${emoji}: `));
+            });
+        }
+    }, [value, updateValue, rootId, cursorPosition]);
+
+    const renderItem = useCallback(({item}: {item: string}) => {
+        const completeItemSuggestion = () => completeSuggestion(item);
+        return (
+            <TouchableWithFeedback
+                onPress={completeItemSuggestion}
+                underlayColor={changeOpacity(theme.buttonBg, 0.08)}
+                type={'native'}
+            >
+                <View style={style.row}>
+                    <View style={style.emoji}>
+                        <Emoji
+                            emojiName={item}
+                            textStyle={style.emojiText}
+                            size={24}
+                        />
+                    </View>
+                    <Text style={style.emojiName}>{`:${item}:`}</Text>
+                </View>
+            </TouchableWithFeedback>
+        );
+    }, [completeSuggestion, theme]);
```

Comment:
In an old PR, I noticed that if you put the dependency as `theme.buttonBg` instead of just `theme` which is the whole object, then, the changes are effectively recomputed.  Comparing one value instead of an object that may have the same reference and thus serving the refreshed function.